### PR TITLE
Fix from_robot overview for Python SDK in docs

### PR DESCRIPTION
**Pull request by sguequierre**
[small fixup I noticed while working on service API docs, was merged in error not update causing drift]

* Switch name to parameter 
* Return resource name
* Add clarifying note for both python and go

### CODE REVIEW

File: docs/program/apis/_index.md [hunk 1]
```diff
@@ -32,12 +32,13 @@ Get the `"name"` of a resource.
...
 
+- `robot` [(RobotClient)](https://python.viam.dev/autoapi/viam/robot/client/index.html#viam.robot.client.RobotClient): The robot.
 - `name` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The `name` of the resource.
 
 **Returns:**
 
-- `robot` [(RobotClient)](https://python.viam.dev/autoapi/viam/robot/client/index.html#viam.robot.client.RobotClient): The robot.
-- `name` [(str)](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.ResourceName): The "name" of the resource.
+- [(Resource)](https://python.viam.dev/autoapi/viam/resource/base/index.html): The named resource if it exists on your robot.
+For example, an [Arm](https://pkg.go.dev/go.viam.com/rdk/components/arm#Arm).
```
Comment (JessamyT) [inline docs/program/apis/_index.md:41]:
```suggestion
For example, an [arm](https://pkg.go.dev/go.viam.com/rdk/components/arm#Arm).
```
[nit]

File: docs/program/apis/_index.md [hunk 2]
```diff
@@ -55,7 +56,8 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
 
 **Returns:**
 
-- [(Resource)](https://pkg.go.dev/go.viam.com/rdk@v0.2.47/resource#Name): Your named resource. For example, an [Arm](https://pkg.go.dev/go.viam.com/rdk/components/arm#Arm).
+- [(Resource)](https://pkg.go.dev/go.viam.com/rdk@v0.2.47/resource#Name): The named resource if it exists on your robot.
+For example, an [Arm](https://pkg.go.dev/go.viam.com/rdk/components/arm#Arm).
```
Comment (JessamyT) [inline docs/program/apis/_index.md:60]:
```suggestion
For example, an [arm](https://pkg.go.dev/go.viam.com/rdk/components/arm#Arm).
```
[nit]

File: docs/program/apis/_index.md
Comment (JessamyT) [inline docs/program/apis/_index.md:1]: Line 28 seems wrong. FromRobot gets a resource, not the name of a resource.

Maybe fix line 82 while you're at it for newline and consistency.

**Review by JessamyT — CHANGES_REQUESTED** [review]
I know line 28 is out of diff but while reviewing this I realized that's totally wrong and maybe you want to fix it as well while we're here.

File: docs/program/apis/_index.md
Comment (sguequierre) [reply docs/program/apis/_index.md:1]: @JessamyT fixed! Does this look better?

Comment (JessamyT) [reply docs/program/apis/_index.md:1]: I think it's a hard thing to word well, but at least it's not actively incorrect now! Will make opt suggestions

File: docs/program/apis/_index.md [hunk 3]
```diff
@@ -77,7 +79,8 @@ For example, a component with [type `arm`](https://ts.viam.dev/classes/ArmClient
 
 **Returns:**
 
-- [(Resource)](https://ts.viam.dev/interfaces/Resource.html): Your named resource. For example, an [ArmClient](https://ts.viam.dev/classes/ArmClient.html).
+- [(Resource)](https://ts.viam.dev/interfaces/Resource.html): Your named resource.
```
Comment (JessamyT) [inline docs/program/apis/_index.md:82]:
```suggestion
- [(Resource)](https://ts.viam.dev/interfaces/Resource.html): The named resource if it exists on your robot.
```
(for consistency with the other tabs)

**Review by JessamyT — APPROVED** [review]
Consistency nit (unless there's a language idiomatic reason for it to be different!). Otherwise looks good to me

**Pull request merged**